### PR TITLE
fix broken `breaks tracking`

### PR DESCRIPTION
**Pull request by maxtaco**
- this never worked. sorry @chrisnojima nojima and @cjb for the bug :(
- add a test, piggy backed on top of SoftSnoozeTest. I asserted by
  hand that the test fails before the bug is fixed.
- also remove some dead code

### CODE REVIEW

File: go/engine/id_test.go [hunk 1]
```diff
@@ -73,13 +73,6 @@ func checkDisplayKeys(t *testing.T, idUI *FakeIdentifyUI, callCount, keyCount in
 			t.Logf("key: %+v, %+v", k, v)
 		}
 	}
-
-	// this doesn't work anymore:
```
Comment (maxtaco) [inline go/engine/id_test.go]: nuke this...

File: go/engine/id_test.go [hunk 2]
```diff
@@ -177,6 +170,7 @@ type FakeIdentifyUI struct {
 	Outcome         *keybase1.IdentifyOutcome
 	StartCount      int
 	Token           keybase1.TrackToken
+	BrokenTracking  bool
```
Comment (maxtaco) [inline go/engine/id_test.go]: the `FakeIdentifyUI` keeps track of if there was a `BreaksTracking` bool or not...

File: go/engine/soft_snooze_test.go [hunk 3]
```diff
@@ -140,6 +142,9 @@ func TestSoftSnooze(t *testing.T) {
 	if pe := libkb.ImportProofError(result.ProofResult); pe == nil {
 		t.Fatal("expected a Rooter error result")
 	}
+	if !idUI.BrokenTracking {
```
Comment (maxtaco) [inline go/engine/soft_snooze_test.go]: We expect the bool to be set. I tested that this check breaks before the bugfix below.

File: go/libkb/rpc_exim.go [hunk 4]
```diff
@@ -52,7 +52,7 @@ func (l LinkCheckResult) Export() keybase1.LinkCheckResult {
 	}
 	if l.remoteDiff != nil {
 		ret.RemoteDiff = ExportTrackDiff(l.remoteDiff)
-		if l.diff.BreaksTracking() {
+		if l.remoteDiff.BreaksTracking() {
```
Comment (maxtaco) [inline go/libkb/rpc_exim.go]: welp!

**Comment by maxtaco** [comment]
@cjb  / @chrisnojima  -- A further note.  The `-s` flag works without bugs, to the best of my understanding...

**Comment by patrickxb** [comment]
LGTM

(also restarted your travis build as the system was shutdown in the middle of the test run)

**Comment by chrisnojima** [comment]
Cool thanks max

**Comment by maxtaco** [comment]
Sorry for the bug and not owning up to it right away!

**Pull request merged**